protocol/service.go

```diff
@@ -6,3 +6,7 @@ package protocol
 // B. The `qos` package handles offchain specs & details.
 // C. This allows the `gateway` package to map multiple Service IDs to a single qos implementation, e.g. all EVM blockchain services can be handled by `qos/evm`.
 type ServiceID string
+
+// ServiceIDNoOp is the Service ID for the NoOp QoS service.
+// This is used when a service ID does not yet have a QoS implementation.
+const ServiceIDNoOp ServiceID = "noop"
```

Comment:
Strong opinion: this does not belong here. `protocol` should not in any way be aware of or care about offchain QoS, and PATH has been designed to maintain this restriction.

Mapping of onchain service IDs, AKA authoritative service IDs, to specific QoS implementations is the responsibility of the `cmd` and/or `config` packages (depending on the desired level of flexibility).

If some sort of validation/authority from the `protocol` package is desired to e.g. validate user config/input, we can extend `gateway/qos.go` file to add a method under `QoSService`, e.g.:
```go
type  QoSService interface {
	QoSContextBuilder
        func ValidServiceIDs() ([]protocol.ServiceID, error)
}
```

Reply:
No problem; I have absolutely zero attachment to this variable living here.

The main point is just for us to have a consistent key value to be able to retrieve the NoOp QoS service from the `gatewayQoSService` map. I'll provide more detail in answer to your next comment.